projects/igniteui-angular/src/lib/drop-down/drop-down.component.spec.ts

```diff
@@ -267,11 +270,115 @@ describe('IgxDropDown ', () => {
             tick();
             fixture.detectChanges();
             expect(list.selectedItem).toEqual(list.items[3]);
+            expect(list.onSelecting.emit).toHaveBeenCalledTimes(1);
             expect(list.onSelection.emit).toHaveBeenCalledTimes(1);
             expect(list.onClosed.emit).toHaveBeenCalledTimes(1);
+            expect(fixture.componentInstance.onSelecting).toHaveBeenCalledTimes(1);
             expect(fixture.componentInstance.onSelection).toHaveBeenCalledTimes(1);
         }));
 
+        it('Should check if selection events return the proper eventArgs', fakeAsync(() => {
+            const fixture = TestBed.createComponent(IgxDropDownTestComponent);
+            fixture.detectChanges();
+
+            const button = fixture.debugElement.query(By.css('button')).nativeElement;
+            const list = fixture.componentInstance.dropdown;
+            const mockObj = jasmine.createSpyObj('mockEvt', ['stopPropagation', 'preventDefault']);
+            spyOn(list.onSelecting, 'emit').and.callThrough();
+            spyOn(list.onSelection, 'emit').and.callThrough();
+            spyOn(fixture.componentInstance, 'onSelecting');
+            spyOn(fixture.componentInstance, 'onSelection');
+            expect(list).toBeDefined();
+
+            button.click(mockObj);
+            tick();
+            fixture.detectChanges();
+            const lastListItem = list.items[3].element.nativeElement;
+
+            lastListItem.click({});
+            tick();
+            fixture.detectChanges();
+            expect(list.selectedItem).toEqual(list.items[3]);
+            expect(list.onSelecting.emit).toHaveBeenCalledTimes(1);
+            expect(list.onSelection.emit).toHaveBeenCalledTimes(1);
+            expect(fixture.componentInstance.onSelecting).toHaveBeenCalledTimes(1);
```

Comment:
I think we don't need to spy on `fixture.componentInstance.onSelecting`.

Reply:
Also you could remove these expects as they are the same as in the previous test.